neonvm-runner/cmd/main.go

```diff
@@ -669,6 +673,85 @@ func main() {
 	}
 }
 
+type NetworkMonitoringMetrics struct {
+	IngressBytes, EgressBytes       prometheus.Counter
+	Errors                          *prometheus.CounterVec
+	IngressBytesRaw, EgressBytesRaw uint64 // Absolute values to calc increments for Counters
+}
+
+func NewMonitoringMetrics(reg *prometheus.Registry) *NetworkMonitoringMetrics {
+	m := &NetworkMonitoringMetrics{
+		IngressBytes: prometheus.NewCounter(
+			prometheus.CounterOpts{
+				Name: "runner_vm_ingress_bytes",
+				Help: "Number of bytes received by the VM from the open internet",
+			},
+		),
+		EgressBytes: prometheus.NewCounter(
+			prometheus.CounterOpts{
+				Name: "runner_vm_egress_bytes",
+				Help: "Number of bytes sent by the VM to the open internet",
+			},
+		),
+		IngressBytesRaw: 0,
+		EgressBytesRaw:  0,
+		Errors: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "runner_vm_network_fetch_errors_total",
+				Help: "Number of errors while fetching network monitoring data",
+			},
+			[]string{"cause"},
+		),
+	}
+	reg.MustRegister(m.IngressBytes)
+	reg.MustRegister(m.EgressBytes)
+	reg.MustRegister(m.Errors)
```

Comment:
Suggestion: Try wrapping each `prometheus.NewFoo(...)` with `util.RegisterMetric(reg, ...)` — we added that to avoid the series of detached `reg.MustRegister(...)` calls that we'd otherwise have to have.